src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1235,10 +1235,33 @@ static int client_receive_message(
         return 0;
 }
 
+static int client_get_lifetime(sd_dhcp6_client *client, uint32_t *lifetime_t1,
+                               uint32_t *lifetime_t2) {
+        uint32_t t1 = 0, t2 = 0;
+
+        assert_return(client, -EINVAL);
+        assert_return(client->lease, -EINVAL);
+
+        if (client->request & DHCP6_REQUEST_IA_NA && client->lease->ia.addresses) {
+                t1 = be32toh(client->lease->ia.ia_na.lifetime_t1);
+                t2 = be32toh(client->lease->ia.ia_na.lifetime_t2);
+        } else if (client->request & DHCP6_REQUEST_IA_PD && client->lease->pd.addresses) {
+                t1 = be32toh(client->lease->pd.ia_pd.lifetime_t1);
+                t2 = be32toh(client->lease->pd.ia_pd.lifetime_t2);
+        } else
+                return -ENOMSG;
+
+        *lifetime_t1 = t1;
+        *lifetime_t2 = t2;
```

Comment:
Not that it matters too much, by why not assign `lifetime_t1` and `lifetime_t2` directly?

Reply:
I apparently thought some other code was needed as well, logic fixed since none added.